pkg/operator/staticpod/controllers.go

```diff
@@ -285,14 +290,18 @@ func (b *staticPodOperatorControllerBuilder) ToControllers() (manager.Controller
 		eventRecorder,
 	), 1)
 
-	manager.WithController(staticpodfallback.New(
-		b.operandNamespace,
-		b.operandPodLabelSelector,
-		b.staticPodOperatorClient,
-		b.kubeInformers,
-		b.enableStartMonitor,
-		b.eventRecorder,
-	), 1)
+	if b.operandPodLabelSelector.Empty() {
+		eventRecorder.Warning("OperandPodLabelMissing", "not enough information provided, not all functionality is present")
```

Comment:
I think we should return an error since we need a way to find the operand.

Reply:
actually, we could do without the selector. At the moment it is used only in the `staticPodFallbackConditionController`. In the worst-case scenario, we would loop over all pods in a given namespace.

Reply:
As I understand the `staticPodFallbackConditionController` is narrowly focused on the static pods/operands. So it needs to ignore any other pod. It could incorrectly set the degraded condition otherwise.

Reply:
not sure why do we want to fire an event in that case? It will make troubleshooting harder. We should either return an error or let it fall through (previous behavior).

Reply:
I do not consider the controller to be crucial. I'd rather set no condition than set it incorrectly. Though, I might be wrong here.